examples/source/style-layout/Dark_Mode_Toggle.html

```diff
@@ -0,0 +1,150 @@
+<!---
+- author: tomayac
+- formats
+  - websites
+  - stories
+--->
+
+<!--
+  ## introduction
+
+  todo: This sample demonstrates how to create a dark mode toggle.
+-->
+<!-- -->
+<!doctype html>
+<html ⚡>
+<head>
+  <meta charset="utf-8">
+  <link rel="canonical" href="{{filename}}">
+  <meta name="viewport" content="width=device-width,minimum-scale=1,initial-scale=1">
+  <script async src="https://cdn.ampproject.org/v0.js"></script>
+  <title>{{title}}</title>
+  <style amp-custom>
+  /* default styles, please remove any you're not using */
+  :root {
+    --color-text-light: #fff;
+    --color-text-dark: #000;
+    --color-bg-light: #FAFAFC;
+    --color-bg-dark: #111;
+    --heading-color: red;
+    --text-color: --color-text-dark;
+    --background-color: --color-text-light;
+    --image-filter: grayscale(0%);
+
+    font-family: sans-serif;
+  }
+
+  #dark-mode:checked ~ label::before {
+    content: "☀️";
+  }
+
+  label[for="dark-mode"]::before {
+    content: "🌒";
+  }
+
+  #dark-mode,
+  label[for="dark-mode"] {
+    color: var(--color-text-dark);
+    position: absolute;
+    top: 1rem;
+    right: 0.25rem;
+  }
+
+  @media (prefers-color-scheme: dark) {
+    :root {
+      --heading-color: turquoise;
+      --text-color: var(--color-text-light);
+      --background-color: var(--color-bg-dark);
+      --image-filter: grayscale(50%);
+    }
+
+    label[for="dark-mode"] {
+      display: none;
+    }
+  }
+
+  #dark-mode,
+  #dark-mode:checked ~ * {
+    --heading-color: turquoise;
+    --text-color: var(--color-text-light);
+    --background-color: var(--color-bg-dark);
+    --image-filter: grayscale(50%);
+
+    color: var(--color-text-light);
+  }
+
+  body, html, .wrapper {
```

Comment:
please don't define global rules (body, html, h1, amp-img, ...) as these might interfere with the site.